instructions/instructions.cc

```diff
@@ -2,24 +2,55 @@
 #include "halting/halting.h"
 #include "store_memory/store_memory.h"
 
+#include <iostream>
+
 using namespace std;
 
 // TODO: Jenny please implement this function!
 // Given a string s, generate the new string where we take the first letter concatenated with
 // every other first letter after a dash. e.g.: "so-much" => "sm", "please-no" => "pn"
 string parseInstructionType(string s) {
-  return "";
+  string abbrev;
+  abbrev = s[0];
+  
+  for(std::string::size_type i = 1; i < s.size(); ++i) { // assumes s can't end with '-' -> need a check?
```

Comment:
yes, you're right! what would be great is if we can add a check, and throw an exception!